x-pack/filebeat/docs/inputs/input-aws-s3.asciidoc

```diff
@@ -0,0 +1,95 @@
+visibility timeout of that sqs message will be reset to make sure the message
+does not go back to the queue in the middle of the processing. If there are
+errors happening during the processing of the s3 object, then the process will be
+stopped and the sqs message will be returned back to the queue.
+
+Example configuration 1:
+
+["source","yaml",subs="attributes"]
+----
+{beatname_lc}.inputs:
+- type: s3
+  queue_url: https://sqs.ap-southeast-1.amazonaws.com/1234/test-s3-queue
+  access_key_id: '${AWS_ACCESS_KEY_ID:""}'
+  secret_access_key: '${AWS_SECRET_ACCESS_KEY:""}'
+  session_token: '${AWS_SESSION_TOKEN:"”}'
+----
+
+Example configuration 2:
+
+["source","yaml",subs="attributes"]
+----
+{beatname_lc}.inputs:
+- type: s3
+  queue_url: https://sqs.ap-southeast-1.amazonaws.com/1234/test-s3-queue
+  credential_profile_name: test-s3-input
+----
+
+The `s3` input supports the following configuration options plus the
+<<{beatname_lc}-input-{type}-common-options>> described later.
+
+[float]
+==== `queue_url`
+
+URL of the AWS SQS queue that messages will be received from. Required.
+
+[float]
+==== `visibility_timeout`
+
+The duration (in seconds) that the received messages are hidden from subsequent
+retrieve requests after being retrieved by a ReceiveMessage request.
+This value needs to be a lot bigger than filebeat collection frequency so
+if it took too long to read the s3 log, this sqs message will not be reprocessed.
+The default visibility timeout for a message is 300 seconds. The minimum
+is 0 seconds. The maximum is 12 hours.
+
+[float]
+==== `aws credentials`
+
+In order to make AWS API calls, `s3` input requires AWS credentials. Users can
+either put the values into the configuration for `access_key_id`,
+`secret_access_key` and/or `session_token`, or use environment variable
+`AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and/or `AWS_SESSION_TOKEN` instead.
+Please use `Example configuration 1` as an example.
+
+Or shared AWS credentials file can be used with `credential_profile_name` as an
+optional configuration. Please use `Example configuration 2` as an example. If
+`credential_profile_name` is not specified, then `s3` input will consume
+credentials from shared AWS credentials file with `default` profile name.
+
+Please see <<aws-credentials-config>> for more details about how to configure aws
+credentials.
+
+=== S3 and SQS setup
+Enable bucket notification: any new object creation in S3 bucket will also
+create a notification through SQS. Please see
+https://docs.aws.amazon.com/AmazonS3/latest/dev/ways-to-add-notification-config-to-bucket.html#step1-create-sqs-queue-for-notification[create-sqs-queue-for-notification]
+for more details.
+
+[id="{beatname_lc}-input-{type}-common-options"]
+include::../../../../filebeat/docs/inputs/input-common-options.asciidoc[]
+
+[id="aws-credentials-config"]
+include::{libbeat-xpack-dir}/docs/aws-credentials-config.asciidoc[]
```

Comment:
When you rebase, you'll pick up some changes that @andrewvc made for Heartbeat. This will require you to add some examples that are specific to Filebeat. He can help you with the details if I'm unavailable.

Reply:
I will do a rebase and fix it then! Thanks!